### ISSUE TITLE
Crashes on EVReflection attempt to recreate CKRecordID based on dictionary of properties

### ISSUE THREAD

**Issue by davidlawson** [evermeer/EVCloudKitDao#63]
I get the following error message when EVCloudData tries to restore data from a cache:

```
Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'You must call -[CKRecordID initWithRecordName:] or -[CKRecordID initWithRecordName:zoneID:]'
```

It looks like the issue was also mentioned here:
https://github.com/evermeer/EVCloudKitDao/issues/39#issuecomment-162350356

**Comment by mashe** [comment]
Looks like that [this commit](https://github.com/evermeer/EVReflection/commit/a1f3f0d3c8c532720d5875a06494487fce9ed84e) broke normal behaviour
I use EVReflection 2.32.1 unless the issue is fixed

**Comment by evermeer** [comment]
I have pushed a fix for this as EVReflection version 2.35.0
